### PR TITLE
Convert markdown issue templates into issue forms

### PR DESCRIPTION
**Pull request by webknjaz**
##### SUMMARY

This change makes the submitted GitHub issue structure predictable.

###### **[:point_right: ISSUE FORMS FROM THIS PR IN THE DEMO REPOSITORY :point_left:](/ansible/issue-forms-demo-for-ansible/issues/new/choose)**

**UPD:** @mkrizek verified that ansibull is capable of digesting issues created with these forms.

##### ISSUE TYPE

- Maintenance Pull Request

##### COMPONENT NAME

.github/ISSUE_TEMPLATE/*

##### ADDITIONAL INFORMATION

https://gh-community.github.io/issue-template-feedback/structured/

### CODE REVIEW

**Review by mattclay — DISMISSED** [review]
We should probably experiment with this in a separate repo so everyone can see how it will work. That will make it easier to comment on and decide if this is something we want.

Also, the checkboxes should be omitted. They aren't part of the original templates.

File: .github/ISSUE_TEMPLATE/documentation_report.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+- type: checkboxes
+  attributes:
+    label: Code of Conduct
+    description: |
+      Read the [Ansible Code of Conduct][CoC] first.
+
+      [CoC]: https://docs.ansible.com/ansible/latest/community/code_of_conduct.html?utm_medium=github&utm_source=issue_form--documentation_report.yml
+    options:
+    - label: I agree to follow the Ansible Code of Conduct
+      required: true
```
Comment (webknjaz) [inline .github/ISSUE_TEMPLATE/documentation_report.yml]: @mattclay are you talking about this :point_up:? Would it be better to add it in a separate PR or are you opposed to the idea in general?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      Verify first that your issue is not [already reported on GitHub][issue search].
+      Also test if the latest release and devel branch are affected too.
+      *Complete **all** sections as described, this form is processed automatically.*
+
+      [issue search]: https://github.com/ansible/ansible/search?q=is%3Aissue&type=issues
+
+
+- type: textarea
+  attributes:
+    label: SUMMARY
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: I've already check, and we can make this less shouty without breaking the bot
```suggestion
    label: Summary
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    placeholder: >-
+      When I try to do X with ansible-core from the devel branch on GitHub, Y
+      breaks in a way Z under the env E. Here are all the details I know
+      about this problem...
+  validations:
+    required: true
+
+- type: dropdown
+  attributes:
+    label: ISSUE TYPE
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
    label: Issue Type
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    # FIXME: Once GitHub allows defining the default choice, update this
+    options:
+    - Bug Report
+  validations:
+    required: true
+
+- type: input
+  attributes:
+    label: COMPONENT NAME
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
    label: Component Name
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    description: >-
+      Write the short name of the module, plugin, task or feature below,
+      *use your best guess if unsure*.
+    placeholder: dnf, apt, yum, pip, user etc.
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: ANSIBLE VERSION
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
    label: Ansible Version
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+      ```console (paste below)
+      $ ansible --version
+
+      ```
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: CONFIGURATION
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
    label: Configuration
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+      ```console (paste below)
+      $ ansible-config dump --only-changed
+
+      ```
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: OS / ENVIRONMENT
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
    label: OS / Environment
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+      Provide all relevant information below, e.g. target OS versions,
+      network device firmware, etc.
+    placeholder: RHEL 8, CentOS Stream etc.
+  validations:
+    required: true
+
+
+- type: textarea
+  attributes:
+    label: STEPS TO REPRODUCE
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
    label: Steps to Reproduce
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+      <!--- Paste example playbooks or commands between quotes below -->
+      ```yaml (paste below)
+
+      ```
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: EXPECTED RESULTS
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
    label: Expected Results
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+      Describe what you expected to happen when running the steps above.
+    placeholder: >-
+      I expected X to happen because I assumed Y and was shocked
+      that it did not.
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: ACTUAL RESULTS
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
    label: Actual Results
```

File: .github/ISSUE_TEMPLATE/documentation_report.yml [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+      Verify first that your issue is not [already reported on GitHub][issue search].
+      Also test if the latest release and devel branch are affected too.
+      *Complete **all** sections as described, this form is processed automatically.*
+
+      [issue search]: https://github.com/ansible/ansible/search?q=is%3Aissue&type=issues
+
+
+- type: textarea
+  attributes:
+    label: SUMMARY
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/documentation_report.yml]:
```suggestion
    label: Summary
```

File: .github/ISSUE_TEMPLATE/documentation_report.yml [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+    placeholder: >-
+      I was reading the ansible-core documentation of version X and I'm having
+      problems understanding Y. It would be very helpful if that got
+      rephrased as Z.
+  validations:
+    required: true
+
+- type: dropdown
+  attributes:
+    label: ISSUE TYPE
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/documentation_report.yml]:
```suggestion
    label: Issue Type
```

File: .github/ISSUE_TEMPLATE/documentation_report.yml [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+    # FIXME: Once GitHub allows defining the default choice, update this
+    options:
+    - Documentation Report
+  validations:
+    required: true
+
+- type: input
+  attributes:
+    label: COMPONENT NAME
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/documentation_report.yml]:
```suggestion
    label: Component Name
```

File: .github/ISSUE_TEMPLATE/documentation_report.yml [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+    description: >-
+      Write the short name of the rst file, module, plugin, task or
+      feature below, *use your best guess if unsure*.
+    placeholder: docs/docsite/rst/dev_guide/debugging.rst
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: ANSIBLE VERSION
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/documentation_report.yml]:
```suggestion
    label: Ansible Version
```

File: .github/ISSUE_TEMPLATE/documentation_report.yml [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+      ```console (paste below)
+      $ ansible --version
+
+      ```
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: CONFIGURATION
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/documentation_report.yml]:
```suggestion
    label: Configuration
```

File: .github/ISSUE_TEMPLATE/documentation_report.yml [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+      ```console (paste below)
+      $ ansible-config dump --only-changed
+
+      ```
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: OS / ENVIRONMENT
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/documentation_report.yml]:
```suggestion
    label: OS / Environment
```

File: .github/ISSUE_TEMPLATE/documentation_report.yml [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+    description: >-
+      Provide all relevant information below, e.g. OS version,
+      browser, etc.
+    placeholder: Fedora 33, Firefox etc.
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: ADDITIONAL INFORMATION
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/documentation_report.yml]:
```suggestion
    label: Additional Information
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+      Verify first that your issue is not [already reported on GitHub][issue search].
+      Also test if the latest release and devel branch are affected too.
+      *Complete **all** sections as described, this form is processed automatically.*
+
+      [issue search]: https://github.com/ansible/ansible/search?q=is%3Aissue&type=issues
+
+
+- type: textarea
+  attributes:
+    label: SUMMARY
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/feature_request.yml]:
```suggestion
    label: Summary
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+    placeholder: >-
+      I am trying to do X with ansible-core from the devel branch on GitHub and
+      I think that implementing a feature Y would be very helpful for me and
+      every other user of ansible-core because of Z.
+  validations:
+    required: true
+
+- type: dropdown
+  attributes:
+    label: ISSUE TYPE
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/feature_request.yml]:
```suggestion
    label: Issue Type
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+    # FIXME: Once GitHub allows defining the default choice, update this
+    options:
+    - Feature Idea
+  validations:
+    required: true
+
+- type: input
+  attributes:
+    label: COMPONENT NAME
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/feature_request.yml]:
```suggestion
    label: Component Type
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+    description: >-
+      Write the short name of the module, plugin, task or feature below,
+      *use your best guess if unsure*.
+    placeholder: dnf, apt, yum, pip, user etc.
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: ADDITIONAL INFORMATION
```
Comment (gundalow) [inline .github/ISSUE_TEMPLATE/feature_request.yml]:
```suggestion
    label: Additional Information
```

**Review by gundalow — COMMENTED** [review]
Thank you for doing this.

**Comment by webknjaz** [comment]
@mattclay I've made a demo in a separate repo as you suggested: https://github.com/ansible/issue-forms-demo-for-ansible/issues/new/choose

File: .github/ISSUE_TEMPLATE/documentation_report.yml (hunk 1 above)
Comment (mattclay) [reply .github/ISSUE_TEMPLATE/documentation_report.yml]: Yes, it's not part of the current templates.

I wouldn't open a separate PR to add it either. The issue templates should contain the minimum text and prompts necessary to facilitate reporting issues. We should review them periodically to see if there is anything we can remove or clarify to encourage better issue reporting. For example:

* The documentation issue probably doesn't need the "OS / Environment" section. I believe @bcoca already mentioned this elsewhere.
* With the way these templates work, the "Additional Information" section is redundant with the comment section that immediately follows.

Comment (webknjaz) [reply .github/ISSUE_TEMPLATE/documentation_report.yml]: > * With the way these templates work, the "Additional Information" section is redundant with the comment section that immediately follows.

WSYWIG is only enabled for one form, for screenshots. It's non-enforceable.

> I wouldn't open a separate PR to add it either.

I checked with @gundalow if it's a good idea to have the CoC check and he thinks it is.

**Review by gundalow — APPROVED** [review]
This is great, thank you for doing it.
